backend/danswer/background/celery/celery_utils.py

```diff
@@ -84,25 +101,36 @@ def should_prune_cc_pair(
     return time_since_last_pruning.total_seconds() >= connector.prune_freq
 
 
+def rate_limit_doc_generation(doc_batch: list[Document]) -> set[str]:
```

Comment:
this name doesn't feel accurate. It doesn't actually do any rate limiting. This is more like `document_batch_to_ids` and the rate limiting happens when wrapping this function